packages/twenty-front/src/modules/object-record/record-merge/components/MergeRecordsFooter.tsx

```diff
@@ -0,0 +1,70 @@
+import { SIDE_PANEL_FOCUS_ID } from '@/command-menu/constants/SidePanelFocusId';
+import { useHotkeysOnFocusedElement } from '@/ui/utilities/hotkey/hooks/useHotkeysOnFocusedElement';
+import styled from '@emotion/styled';
+import { Key } from 'ts-key-enum';
+import { IconArrowMerge, IconDotsVertical } from 'twenty-ui/display';
+import { Button } from 'twenty-ui/input';
+
+const StyledFooterContainer = styled.div`
+  align-items: flex-end;
+  background: ${({ theme }) => theme.background.primary};
+  border-top: 1px solid ${({ theme }) => theme.border.color.light};
+  display: flex;
+  gap: ${({ theme }) => theme.spacing(2)};
+  justify-content: flex-end;
+  padding: ${({ theme }) => theme.spacing(3)};
+`;
+
+const StyledFooterActions = styled.div`
+  display: flex;
+  align-items: flex-end;
+  gap: ${({ theme }) => theme.spacing(2)};
+`;
+
+type MergeRecordsFooterProps = {
+  onMerge: () => void;
```

Comment:
we should avoid this code pattern where we pass call backs to components. This leads to callbacks drilldown.

This is only acceptable in case we have a UI behavior (dropdown onOpen for instance) which is very generic. Here this seems to be a business related code. In this case, prefer using a hook (useMergeRecord for instance) and call it directly where you need

Reply:
it's very unlikely that we re-use MergeRecordsFooter somewhere else with a different callback. If we do, it's probably that we are having the wrong abstraction and that we should introduce another MergeRecordsSpecificFooter that compose with smaller components